vyper/parser/parser_utils.py

```diff
@@ -52,16 +56,25 @@
 # Get a decimal number as a fraction with denominator multiple of 10
 def get_number_as_fraction(expr, context):
     context_slice = context.origcode.splitlines()[expr.lineno - 1][expr.col_offset:]
-    t = 0
-    while t < len(context_slice) and context_slice[t] in '0123456789.':
-        t += 1
-    if t < len(context_slice) and context_slice[t] == 'e':
-        raise InvalidLiteralException("Literals in scientific notation not accepted.")
-    top = int(context_slice[:t].replace('.', ''))
-    bottom = 1 if '.' not in context_slice[:t] else 10**(t - context_slice[:t].index('.') - 1)
-
-    if expr.n < 0:
-        top *= -1
+    # Get the largest string that converts to decimal
+    t = len(context_slice)
+    while t > 0:  # Breaks if we exhaust the context
+        try:
+            literal = Decimal(context_slice[:t])
+            break  # We got our result!
+        except InvalidOperation:  # Only except our Decimal syntax exception
+            # If it does not convert, shrink the context and try again
+            t -= 1
+    if t <= 0:  # We've exhausted the context
+        raise InvalidLiteralException(
```

Comment:
Note, this should actually be a `CompilerPanic` because this condition should never occur as long as this function is used correctly by other parts of the codebase.